analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -908,3 +1001,58 @@ struct StructWithValuePropertyAndCastOperators
     }
 }
 
+namespace TypeWithValueStaticProperty
+{
+    class Test
+    {
+        void Basics()
+        {
+            _ = AClassWithStaticValueProperty.Value;                                // Compliant, not on nullable value type
```

Comment:
You need to learn "Null" for all the properties first. Otherwise, you would never get the rule triggered:
```suggestion
            _ = AClassWithStaticValueProperty == null ? AClassWithStaticValueProperty.Value : null;                                // Compliant, not on nullable value type
```

Reply:
[DRAFT]

A bit of context: these tests have been written in the context of https://github.com/SonarSource/sonar-dotnet/pull/7000, to solve the `NullReferenceException` instances encountered on peach. The issue was due to custom `Value` properties (user-defined, not `Nullable<T>.Value`) generating `PropertyReferenceOperationWrapper` instances having `Instance` equal to `null`, and that is because they are `static`.

This PR is still in Draft state, assigned to me on the "In Progress" column of the Kanban board: not yet ready to be reviewed because not yet rebased onto the latest `feature/SE`. As a result of that, it now contains commits that have been reviewed and merged into `feature/SE`. So a good chunk of reviewed code has already been reviewed in https://github.com/SonarSource/sonar-dotnet/pull/7000 and merged into the `feature/SE` branch.

Now, coming to the specific problem: since these tests were written to reproduce and fix NRE on peach, they don't aim at reproducing non-compliance. By the way, that would be hard to get for chained `.Value` dereferences anyway, since we don't learn on those properties (since they are not auto-properties).

Scenarios about learning on mutable fields have already been tested in the `MutableField` class, and I can add a dedicated class on `MutableProperties`, to test auto-properties, if you feel that is necessary. My understanding is that, however, those tests would have the correct behavior only after https://github.com/SonarSource/sonar-dotnet/pull/7007 is merged into `feature/SE`, because we don't currently learn from auto-properties.

So, for this PR specifically, I have added